### PR TITLE
Westy dbexceptions

### PR DESCRIPTION
**Pull request by westyUMass**

### CODE REVIEW

File: test/edu/umass/cs/gnsclient/client/singletests/SelectGeoTest.java [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+import org.junit.FixMethodOrder;
+import org.junit.Test;
+import org.junit.runners.MethodSorters;
+
+/**
+ * Functionality test for core elements in the client using the UniversalGnsClientFull.
+ *
+ */
+@FixMethodOrder(MethodSorters.NAME_ASCENDING)
+public class SelectGeoTest {
```
Comment (akarthik10) [inline test/edu/umass/cs/gnsclient/client/singletests/SelectGeoTest.java]: Shouldn't it use `DefaultGNSTest` as we're now migrating everything to it?

Comment (westyUMass) [inline test/edu/umass/cs/gnsclient/client/singletests/SelectGeoTest.java]: Yes, but I'm doing all those changes together as part of MOB-1084. This update is in service of MOB-1079 which is trying to find a bug. The addition of this file is to make a simple reproducible test case that  isolates the cause as much as possible. Although it probably doesn't make a difference in this case my general philosophy is to not introduce another variable (changing the existing code) until I've figure out what the problem is unless I think it's directly related to fixing the problem.

**Comment by westyUMass** [comment]
Cool. Thanks.

**Comment by v-arun** [comment]
Westy, can you provide more detailed commit notes? Did you fix the DB bugs? If so, what was the problem and the fix?

**Comment by westyUMass** [comment]
Not fixed yet. I enhanced the error messages, but I haven't yet replicated the error. I'm getting other errors before it happens that are related to huge records in Paxos. I'll post some results tomorrow. My guess is that the DB error is happening after many runs it is because the test method is not cleaning up after itself.

**Comment by v-arun** [comment]
Ok, that sounds good. But I don't see the point of merging into master changed or new code that is known to be invariant-incompliant. I understand your previous response but want to keep our commit protocol simple--no invariant-incompliant changed files. The general principle is that invariant-incompliant code is not worth something, but worth nothing at all, so you might as well make it compliant already. Or experiment in a branch.

**Comment by v-arun** [comment]
I take my comment back (but not the principle) because DefaultGNSTest inheritance is not yet anointed as an invariant. I'd like it to be though.

**Comment by westyUMass** [comment]
Also I updated MOB-1079 with some status info when I made the commit.

**Pull request merged**